guardian/cdk
# feat: Use LaunchTemplates instead of LaunchConfiguration

## What does this change?

This change moves `GuAutoScalingGroup` to use `LaunchTemplates` as `LaunchConfigurations` [are deprecated](https://aws.amazon.com/blogs/compute/amazon-ec2-auto-scaling-will-no-longer-add-support-for-new-ec2-features-to-launch-configurations/).

The API to the consumer should remain unchanged, but this will results in EC2 instances getting a different `Name` tag in its current implementation. See https://github.com/guardian/cdk/pull/1731/files#diff-b219fec113d015feaadba12ad710f805f4ba8fc1fcf3c270be34c8fd31b67c0fR101 for the logic that generates the `Name`.

> **Note**
> This change removes tests for `Name` tag on an `AutoScalingGroup` as these are not generated or set by GuCDK and moves from the ASG as Tags to `TagSpecifications` on `LaunchTemplate`.

## How to test

See https://github.com/guardian/cdk-playground/pull/344 for a test on a real-world application.

## Checklist

~- [ ] I have listed any breaking changes, along with a migration path [^1]~
~- [ ] I have updated the documentation as required for the described changes [^2]~

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?
